stable/weave-scope/charts/weave-scope-agent/templates/_helpers.tpl

```diff
@@ -58,4 +58,15 @@ Create the name of the service account to use
 {{- else -}}
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Return the apiVerion of daemonset.
+*/}}
+{{- define "daemonset.apiVersion" -}}
+{{- if semverCompare "<1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- print "extensions/v1beta1" -}}
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```

Comment:
How about this?

```suggestion
{{- else -}}
```

In other words, use `apps/v1` as a _default_ and only regress to `extensions/v1beta1` if K8s version is old.

Reply:
Also, I'd check for `1.9` instead of `1.14`, the docs you provided say that `apps/v1` is available since v1.9.